Grand UI Redesign

horazont opened the pull request:
Current state can be viewed on: https://beta.search.jabber.network/rooms/1

Fixes #22, #25, #27, #28, #29, #30, #33.

Thanks to @nkreer for the accessibility review.

horazont committed:
Mark avatar as aria-hidden to prevent screenreaders from reading the dummy letter

horazont committed:
Let’s try this tab thing

horazont committed:
Clean up CSS

- Remove cruft
- Roughly re-organize into sections

horazont committed:
Re-design room cards

- Orient more closely on standard search engine output format
- Move some text into screenreader-only realm
- Bring number of online users below room name for screen readers
  and text-only browsers
- Adapt font sizes and layout a bit
- Sans-serif only now

Fixes #28.

horazont committed:
Remove titles from SVG icons

Those show up as tooltips in some browsers (e.g. firefox), which
is weird and context-free.

horazont committed:
Make hover background more intense

horazont committed:
Re-design the entire coarse layout

- Save vertical space by condensing the header
- Use a proper colour palette
- Bring some depth to the design

Fixes #30, #29, #25. Initial work on #27.

horazont committed:
De-saturate neutral colors

horazont committed:
Re-add subtitle for context

horazont committed:
Show search terms in page title

Fixes #33.

horazont committed:
Make header bar dark again

Some folks pointed out that it is kind of part of the distinctive
look of the service, so let’s keep it.

Feels odd to me because dark header is the yaml.de default and
most (if not all) of my pages have it 😅

horazont committed:
Move additional search controls into header bar

This makes for a more compact layout and it is sensible to have
them there given that they directly affect the search box function.

horazont committed:
Generalise response caching mechanism

This will be useful once we’ll send static-y stuff which isn’t
text/html or isn’t coming from templates.

(Where "static" means: does not change while the application is
running, but may depend on configuration.)

horazont committed:
Add favicon variants

Fixes #27.

horazont committed:
Make actions on result cards buttonish

This should make them stand out more, and clearly indicate that
it’s an action instead of just a navigation thing. Does it make
sense? I don’t know.

horazont committed:
Relabel View logs -> View history

That’s maybe clearer?

horazont committed:
JID Un-Escape JIDs shown in the web interface

This does not affect JIDs in URLs or as copied to the clipboard.

We are copying the implementation from aioxmpp since there’s not
yet a released version which includes jid_unescape.

Fixes #22.

horazont committed:
Accessibility improvements

horazont committed:
Add info bit about the requirements for join / view history

horazont committed:
Add some (collapsible) whitespace between users icon and number

Makes it look less crammed, but won’t make the number and icon
spill over the borders because it can be collapsed.

horazont committed:
Extend legal notes

horazont committed:
Remove unused CSS files

horazont committed:
Update privacy policy

horazont committed:
Use aria-label for next/prev buttons and move :after into <a/>

This is an attempt to prevent screen readers from reading the
arrow-y brackets.

Thanks to @nkreer for the accessibility review.

horazont committed:
Use h3 for room card headings

This gives a more consistent document view for screenreader
navigation.

Thanks to @nkreer for the accessibility review.

horazont committed:
Remove "go to" part of aria-label on pagination

It is redundant because it’s already a link; it should be clear
that it brings you to another page.

Thanks to @nkreer for the accessibility review.

horazont committed:
Set language attribute on room text, based on room language

Possibly not ideal, but better than nothing to screenreaders.

Thanks to @nkreer for the accessibility review.

horazont committed:
Add h2 in front of footer navigation

For easier navigation with screenreaders.

Thanks to @nkreer for the accessibility review.

horazont committed:
Convert footer navigation subheadings to h3

Again, makes more sense to go from h2 -> h3 instead of -> h6.

Thanks to @nkreer for the accessibility review.

horazont committed:
Add an aria-label to the otherwise confusing ellipsis

Thanks to @nkreer for the accessibility review.

KreerC commented:
Invalid markup is currently generated in room headings, which might prevent screenreaders from seeing the language attributes:

```html
<h3 class="title" "="" lang="de">
```

Also, only one language should be specified for those attributes, unlike

```html
<h3 class="title" lang="'de', 'en'" "="">
```

I also think that the description text for rooms need to be wrapped in paragraph-tags that carry the language tag appropriately in addition to the div container.

horazont commented:
> Also, only one language should be specified for those attributes, unlike
> 
> ```
> <h3 class="title" lang="'de', 'en'" "="">
> ```

I don’t have control over that, unfortunately. The data is provided by the rooms. The only thing I could do is filter out blatantly incorrect language tags (i.e. anything which isn’t two or three letters) and omit the language tag on those rooms.

> I also think that the description text for rooms need to be wrapped in paragraph-tags that carry the language tag appropriately in addition to the div container.

That makes sense. I should use ``<p/>`` instead of the ``<div/>``.

horazont commented:
While we’re at it, is there anything which is semantically more sensible than the ``<div/>`` I’ve been using for the room address line?

horazont committed:
Fix vertical alignment of language icon

horazont committed:
Use <p/> instead of <div/> for room card text

It’s more semantic and clear.

Thanks to @nkreer for the accessibility review.

horazont committed:
De-saturate the neutral grays more

They have been identified as brown by at least one user and by
myself on my phone, so they’re not chosen well.

horazont committed:
Use the primary instead of the accent colour for the search button

This is more in line with the remainder of the design.
